xwiki-platform-distribution/xwiki-platform-distribution-migrations/src/main/java/org/xwiki/migrations/internal/XWikiPropertiesMetaFieldCleanupTaskConsumer.java

```diff
@@ -0,0 +1,87 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.migrations.internal;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.index.IndexException;
+import org.xwiki.index.TaskConsumer;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+import static com.xpn.xwiki.internal.mandatory.XWikiPreferencesDocumentInitializer.LOCAL_REFERENCE;
+
+/**
+ * Remove the content of the {@code meta} field from {@code XWiki.XWikiProperties} when requested by
+ * {@link R170400000XWIKI23160DataMigration}. Note that this task is internal and shouldn't be used outside of
+ * {@link R170400000XWIKI23160DataMigration}.
+ *
+ * @version $Id$
+ * @since 17.4.0RC1
+ */
+@Component
+@Named(XWikiPropertiesMetaFieldCleanupTaskConsumer.TASK_NAME)
+@Singleton
+public class XWikiPropertiesMetaFieldCleanupTaskConsumer implements TaskConsumer
+{
+    static final String TASK_NAME = "xwiki-properties-meta-field-cleanup";
+
+    private static final String META_FIELD = "meta";
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public void consume(DocumentReference documentReference, String version) throws IndexException
+    {
+        XWikiContext context = this.contextProvider.get();
+        XWiki wiki = context.getWiki();
+        XWikiDocument document;
+        try {
+            document = wiki.getDocument(documentReference, context);
+        } catch (XWikiException e) {
+            throw new IndexException("Unable to retrieve document [%s]".formatted(documentReference), e);
+        }
+        DocumentReference xwikiPreferencesDocumentReferenceNoLocal =
+            new DocumentReference(LOCAL_REFERENCE, context.getWikiReference());
+
+        BaseObject xObject = document.getXObject(xwikiPreferencesDocumentReferenceNoLocal);
+        if (xObject != null && StringUtils.isNotEmpty(xObject.getStringValue(META_FIELD))) {
+            XWikiDocument documentClone = document.clone();
+            documentClone.getXObject(xwikiPreferencesDocumentReferenceNoLocal).setStringValue(META_FIELD, "");
```

Comment:
Since that TaskConsumer is supposed to be triggered only when the document does need to be modified, it would probably make the code a bit more readable to directly clone the document and avoid the double getXObject.